editor/src/clj/editor/app_view.clj

```diff
@@ -294,6 +296,81 @@
                                 (get sub-selections-by-resource-node active-resource-node)))
   (output refresh-tab-panes g/Any :cached (g/fnk [^SplitPane editor-tabs-split open-views open-dirty-views]
                                             (let [tab-panes (.getItems editor-tabs-split)]
+
+                                              ;TODO: Extract this logic to another method.
+                                              (let [total-tabs (reduce + (map #(-> ^TabPane % (.getTabs) (.size)) tab-panes))
+                                                    is-empty (zero? total-tabs)
+                                                    quick-help-box (.lookup (.getParent editor-tabs-split) "#quick-help-box")
+                                                    ^GridPane box-items (.lookup (.getParent editor-tabs-split) "#quick-help-items")
+                                                    is-valid (and quick-help-box box-items)]
+                                                (if is-valid (.setVisible quick-help-box is-empty))
+                                                (if (and is-valid (zero? (.size (.getChildren box-items))))
+                                                ;(if (and is-valid)
+                                                  (do
+                                                    (.setVisible quick-help-box is-empty)
+
+                                                    ;TODO: Do nothing if the box-items has children
+                                                    ;TODO: Move the items to the other place where we can define shortcut for each OS
+                                                    (let [items [{:label "Re-Open Closed File" :keys ["Ctrl" "Shift" "T"]}
```

Comment:
I also want to extract these config.
Where is the good place to put this one?
I think we also need dynamic config (by platform). Because each platform may have different shortcut.

Reply:
Users can override the keymap using a config file, so we can't hardcode these.

The `AppView` node you're editing has access to the keymap. Add `keymap-config` as an input dependency to the surrounding `g/fnk`. This ensures the view will be updated if the keymap changes when we get around to making a keymap editor in the future.
```
(g/fnk [^SplitPane editor-tabs-split open-views open-dirty-views keymap-config]
  ...)
```
You can then do something like:
```
(let [command->key-combo
      (into {}
            (mapcat (fn [[key-combo command-infos]]
                      (map (fn [command-info]
                             [(:command command-info) key-combo])
                           command-infos)))
            keymap)]
  ;; You can now get the key-combo info for each command, i.e.
  (command->key-combo :reopen-recent-file)
  (command->key-combo :open-asset)
  ...)
```
The key-combo is a map that looks like this:
```
{:key "T",
 :alt-down? false,
 :control-down? false,
 :meta-down? true,
 :shift-down? true,
 :shortcut-down? false}
```
`:shortcut-down?` will be `false`, and should be ignored. On macOS, **Cmd** (`:meta-down?`) is used instead of **Ctrl** for basic shortcuts, but **Ctrl** may be used for some advanced shortcuts. You can assume `:meta-down?` will be `false` on non-mac platforms.

Reply:
I used your suggestion. For now, the key-map can be dynamic display by OS.